sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/DoNotShiftByZeroOrIntSize.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+
+namespace Tests.Diagnostics
+{
+    class Foo
+    {
+        private void Test()
+        {
+            byte b = 1 << 10; // Noncompliant {{Either promote shift target to a larger integer type or shift by 2 instead.}}
+//                   ^^^^^^^
+            b = 1 << 10; // Noncompliant {{Either promote shift target to a larger integer type or shift by 2 instead.}}
+//              ^^^^^^^
+            b = 1 << 0; // Noncompliant {{Remove this useless shift by 0.}}
+
+            sbyte sb = 1 << 10; // Noncompliant
+
```

Comment:
Let's add some tests with extra parenthesis because I think your code is not going to work.